app/src/sharedTest/java/org/oppia/android/app/classroom/ClassroomListFragmentTest.kt

```diff
@@ -0,0 +1,878 @@
+    )
+    logIntoAdminTwice()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_HEADER_TEST_TAG).onChildAt(0)
+      .assertTextContains(context.getString(R.string.recommended_stories))
+      .assertIsDisplayed()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_TEST_TAG).onChildAt(0)
+      .assertTextContains("Fifth Exploration")
+      .assertTextContains("SECOND TEST TOPIC")
+      .assertTextContains("SCIENCE")
+      .assertIsDisplayed()
+  }
+
+  @Test
+  // TODO(#5344): Add tests for coming soon topics list.
+  fun testFragment_markAtLeastOneStoryCompletedForAllTopics_displaysComingSoonTopicsList() {
+  }
+
+  @Test
+  // TODO(#5344): Add tests for coming soon topics list.
+  fun testHomeActivity_markFullProgressForSecondTestTopic_displaysComingSoonTopicsText() {
+  }
+
+  /*
+   * # Dependency graph:
+   *
+   *      Fractions
+   *         |
+   *        |
+   *       v
+   * Test topic 0                     Ratios
+   *    \                              /
+   *     \                           /
+   *       -----> Test topic 1 <----
+   *
+   * # Logic for recommendation system
+   *
+   * We always recommend the next topic that all dependencies are completed for. If a topic with
+   * prerequisites is completed out-of-order (e.g. test topic 1 above) then we assume fractions is
+   * already done. In the same way, finishing test topic 2 means there's nothing else to recommend.
+   */
+  @Test
+  fun testFragment_markStory0DonePlayStory1FirstTestTopic_playFractionsTopic_orderIsCorrect() {
+    fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_UPTIME_MILLIS)
+    storyProgressTestHelper.markCompletedTestTopic0Story0(
+      profileId = profileId,
+      timestampOlderThanOneWeek = false
+    )
+    storyProgressTestHelper.markInProgressSavedFractionsStory0Exp0(
+      profileId = profileId,
+      timestampOlderThanOneWeek = false
+    )
+    storyProgressTestHelper.markInProgressSavedTestTopic1Story0(
+      profileId = profileId,
+      timestampOlderThanOneWeek = false
+    )
+    logIntoAdminTwice()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_HEADER_TEST_TAG).onChildAt(0)
+      .assertTextContains(context.getString(R.string.stories_for_you))
+      .assertIsDisplayed()
+
+    composeRule.onNodeWithTag(PROMOTED_STORY_LIST_TEST_TAG).apply {
+      onChildAt(0)
+        .assertTextContains("Fifth Exploration")
+        .assertTextContains("SECOND TEST TOPIC")
+        .assertTextContains("SCIENCE")
+        .assertIsDisplayed()
+
+      onChildAt(1)
+        .assertTextContains("What is a Fraction?")
+        .assertTextContains("FRACTIONS")
+        .assertTextContains("MATHS")
+        .assertIsDisplayed()
+
+      // TODO(#5344): This story should be promoted.
+      onChildAt(2)
+        .assertDoesNotExist()
+      /*.assertTextContains("What is a Ratio?")
+      .assertTextContains("RATIOS AND PROPORTIONAL REASONING")
+      .assertTextContains("MATHS")
+      .assertIsDisplayed()*/
+    }
```

Comment:
Is this to be done in this PR?

Reply:
Right, keeping the todo as it is, will be done in the recommendation logic PR.

Reply:
Let's not check in commented out code. You can use the TODO to make a clear note of the updates needed to this test, but remove the commented out assertions.

Reply:
Removed the commented code and updated the TODO.